### PR TITLE
[Feature] Add RESTful endpoints for judge data in mongo

### PR DESCRIPTION
**Pull request by alyssamui**
Issue #111 

Changelog:
- Added endpoints for `judges`, `rooms`, `rotationTimes`, and `teams `data in the `Judging `MongoDB cluster
- Add controller, service, dao, schema, and models for each table in the `Judging `MongoDB cluster
- Added unit tests for each of the controller functions using jest and a mock MongoDB server
- Updated server (`app.ts`) to connect to the MongoDB Cluster and created routes for each endpoint

Testing:
Unit tests can be run by running `npm test` in the `judging-algorithm` folder

### CODE REVIEW

**Review by mike10911 — APPROVED** [review]
WOW, THIS IS TOP-TIER!!! Everything looks great and works! See Slack for comment about envs

File: backend/judging-algorithm/data/test/db.ts
Comment (NelsonDong41) [inline backend/judging-algorithm/data/test/db.ts:1]: Just being nitpicky but it looks like mongoose.connection.dropDatabase() also deletes all collections and documents. The clearDatabase() function is then unneeded. (https://mongoosejs.com/docs/api/connection.html#Connection.prototype.dropDatabase())

**Review by NelsonDong41 — APPROVED** [review]
Asides from the small comment, all looks good to me.

**Review by jmsargent10 — APPROVED** [review]
LGTM!

File: backend/judging-algorithm/data/controllers/rotationTimes-controller.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import * as rotationTimesDao from "../dao/rotationTimes-dao.js";
+
+  const getRotationTime = async (_req: any, res: any) => {
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/controllers/rotationTimes-controller.ts:3]: nit: shift tabline 3-36

File: backend/judging-algorithm/data/controllers/teams-controller.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import * as teamsDao from "../dao/teams-dao.js";
+
+  const getTeam = async (_req: any, res: any) => {
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/controllers/teams-controller.ts:3]: nit: shift tab lines 3-37

File: backend/judging-algorithm/data/dao/rotationTimes-dao.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { RotationTime } from "../../types.js";
+import rotationTimesModel from "../models/rotationTimes-models.js";
+
+export const getRotationTime = async () => 
+  await rotationTimesModel.find();
+
+export const getRotationTimeById = async (rotationTimeID: string) => 
+  await rotationTimesModel.find({_id: rotationTimeID});
+
+export const getRotationTimeByName =async (rotationTimeName: string) =>
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/dao/rotationTimes-dao.ts:10]: nit: space before async

File: backend/judging-algorithm/data/schemas/judges-schema.ts [hunk 4]
```diff
@@ -3,7 +3,11 @@ import mongoose from "mongoose";
 export const judgesSchema = new mongoose.Schema(
   {
     name: { type: String, required: true },
+    inPerson: { type: Boolean, required: true}
+  },
+  { 
+    collection: "Judges",
+    versionKey: false
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/schemas/judges-schema.ts:10]: why's versionKey required?

File: backend/judging-algorithm/data/test/db.ts
Comment (surbhi-gulati) [reply backend/judging-algorithm/data/test/db.ts:1]: nice catch!

File: backend/judging-algorithm/data/test/rooms-controller.test.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { mockResponse, testCreateRoomRequest } from './test-constants.js';
+
+beforeAll(async () => await db.connectDatabase())
+afterAll(async () => {
+    await db.clearDatabase();
+    await db.closeDatabase();
+})
+
+// create room
+it("Test create room", async () => {
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/test/rooms-controller.test.ts:12]: nit: all page comments can be deleted

File: backend/judging-algorithm/data/test/rotationTimes-controller.test.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import controller from '../controllers/rotationTimes-controller.js'
+import { mockResponse, testCreateRotationTimeRequest } from './test-constants.js';
+
+beforeAll(async () => await db.connectDatabase())
+afterAll(async () => {
+    await db.clearDatabase();
+    await db.closeDatabase();
+})
+
+// create rotation time
```
Comment (surbhi-gulati) [inline backend/judging-algorithm/data/test/rotationTimes-controller.test.ts:11]: nit: all page comments can be deleted. and restructure test names to 
describe("Component") , it ("behavior") style here and in all test suites

**Review by surbhi-gulati — CHANGES_REQUESTED** [review]
GREAT JOB BOTH OF YOU!! I have no content change requests, just formatting nitpicks, <10 mins!

File: backend/judging-algorithm/data/schemas/judges-schema.ts (hunk 4 above)
Comment (alyssamui) [reply backend/judging-algorithm/data/schemas/judges-schema.ts:10]: without versionKey set to false, if we create objects for our database it adds an additional versionKey field in the data

Comment (surbhi-gulati) [reply backend/judging-algorithm/data/schemas/judges-schema.ts:10]: Ohh ok, thanks!!

**Review by surbhi-gulati — APPROVED** [review]
Great job!

**Pull request merged**